mne/label.py

```diff
@@ -897,6 +903,137 @@ def write_label(filename, label, verbose=None):
     return label
 
 
+def _prep_label_split(label, subject=None, subjects_dir=None):
+    """Helper to get label and subject information prior to label spliting"""
+
+    # If necessary, find the label
+    if isinstance(label, BiHemiLabel):
+        raise TypeError("Can only split labels restricted to one hemisphere.")
+    elif isinstance(label, string_types):
+        label = read_label(label)
+
+    # Find the subject
+    subjects_dir = get_subjects_dir(subjects_dir, raise_error=True)
+    if label.subject is None and subject is None:
+        raise ValueError("The subject needs to be specified.")
+    elif subject is None:
+        subject = label.subject
+    elif label.subject is None:
+        pass
+    elif subject != label.subject:
+        raise ValueError("The label specifies a different subject (%r) from "
+                         "the subject parameter (%r)."
+                         % label.subject, subject)
+
+    return label, subject, subjects_dir
+
+
+def _assign_split_membership(label_divs, neighbors):
+    """Helper to assign membership of a list of nodes to label components"""
+
+    # Find overlap between new node neighbors and existing label divisions
+    membership = []
+    for di, div in enumerate(label_divs):
+        if len(neighbors.intersection(div)) > 0: membership.append(di)
+
+    # If node connects to an existing div, assign it to that div
+    if len(membership) > 0:
+        label_divs[membership[0]].update(neighbors)
+
+        # If node connects to 2+ divs, combine them all into one division
+        # (Could also do this once after all nodes memberships are compiled)
+        if membership > 1:
+            for mi in sorted(membership[1:], reverse=True):
+                label_divs[membership[0]].update(label_divs.pop(mi))
+
+    # If edge doesn't connect to existing div, create a new one
+    else:
+        label_divs.append(neighbors)
+
+
+def split_label_contig(label_to_split, subject=None, subjects_dir=None):
```

Comment:
Should I make the related `split_label` function private too? Or is it not worth going through the  deprecation cycle?